### PR TITLE
test: not able to view files on chpl download page when logged in

### PR DESCRIPTION
**Pull request by iyerashwin14**

### CODE REVIEW

File: src/test/resources/Features/login.feature [hunk 1]
```diff
@@ -2,4 +2,4 @@ Feature: User login
 
 Scenario: User logs in as ACB Admin
     Given I'm logged in as an ACB
-    Then I should be logged in to CHPL as ACB Admin
+    Then I should be logged in to CHPL as "ACB Admin"
```
Comment (andlar) [inline src/test/resources/Features/login.feature]: If the idea is to test logging in as different users, the do it correctly:
```
Scenario Outline: User logs in as given ROLE
  Given I'm logged in as "<ROLE>"
  Then I should be logged in to CHPL as "<Name">
Examples
| ROLE       | Name      |
| ROLE_ADMIN | AQA Admin |
| ROLE_ONC   | AQA ONC   |
| ROLE_ACB   | AQA ACB   |
```

File: src/test/resources/Features/OCD-1764-chpldownload.feature [hunk 2]
```diff
@@ -10,3 +9,10 @@ Feature: OCD-1764 - Chpl data files are downloadable and have respective definit
     Given I am on download the CHPL resources page on "DEV"
     When user selects a file in download file box
     Then definition file shows based on download file selection
+
+  Scenario: Admin user sees eight download files
+    Given I'm logged in as an Admin
+    And I should be logged in to CHPL as "Admin"
```
Comment (andlar) [inline src/test/resources/Features/OCD-1764-chpldownload.feature]: This `And` step should not be here. This step is a validation step and not appropriate for this scenario

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java [hunk 3]
```diff
@@ -21,6 +21,8 @@
     private String url = System.getProperty("url");
     private String username = System.getProperty("username");
     private String password = System.getProperty("password");
+    private String adminUsername = System.getProperty("adminUsername");
+    private String adminPassword = System.getProperty("adminPassword");
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java]: All of our users are Admin in one way of another. If you're going to distinguish between them, use the ROLEs we use everywhere else. To get all three of the AQA users in this property file you should have
```
roleAdminUsername
roleAdminPassword
roleOncUsername
roleOncPassword
roleAcbUsername
roleAcbPassword

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java [hunk 4]
```diff
@@ -54,10 +62,26 @@ public void logInAsAcb() {
...
+    @Then("^I should be logged in to CHPL as \"([^\"]*)\"$")
+    public void verifyLoginWasSuccessful(final String role) {
         String actualString = LoginPage.welcomeText(driver).getText();
         assertTrue(actualString.contains("Welcome"));
     }
+
+    /**
+     * Log the user in as an ACB.
+     */
+    @Given("^I'm logged in as an Admin$")
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java]:
```suggestion
    @Given("^I'm logged in as an \"([^\"]*)\"$")
```

Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java]:
```suggestion
    @Given("^I'm logged in as "ROLE"$")
```

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java [hunk 5]
```diff
@@ -54,10 +62,26 @@ public void logInAsAcb() {
...
+    public void logInAsAdmin() {
+        driver.get(url + "#/admin/dpManagement");
+        WebDriverWait wait = new WebDriverWait(driver, TIMEOUT);
+        wait.until(ExpectedConditions.visibilityOf(LoginPage.username(driver)));
+        LoginPage.username(driver).sendKeys(adminUsername);
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java]: Here, do a switch or if/else based on the ROLE string to figure out what username and password you should use:
```
String username;
String password
if (role.equalsIgnoreCase("ROLE_ADMIN")) {
username = roleAdminUsername
... etc.
```

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java [hunk 6]
```diff
@@ -54,10 +62,26 @@ public void logInAsAcb() {
...
     /**
      * Verify login attempt was successful.
+     * @param role as ACB or Admin
      */
-    @Then("^I should be logged in to CHPL as ACB Admin$")
-    public void verifyLoginWasSuccessful() {
+    @Then("^I should be logged in to CHPL as \"([^\"]*)\"$")
+    public void verifyLoginWasSuccessful(final String role) {
         String actualString = LoginPage.welcomeText(driver).getText();
         assertTrue(actualString.contains("Welcome"));
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java]: This "Welcome" text should be extended to show something like "Welcome, <insert appropriate "name" variable here>"

**Review by andlar — CHANGES_REQUESTED** [review]

**Review by ashwinimore — COMMENTED** [review]
We did talk about most of the suggested changes. I am ok with the PR once Andrew's comments are resolved.

**Review by ashwinimore — APPROVED** [review]

File: src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java [hunk 7]
```diff
@@ -14,50 +13,73 @@
...
+        if (role.equalsIgnoreCase("ROLE_ADMIN")) {
+            username = roleAdminUsername;
+            password = roleAdminPassword;
+        } else if (role.equalsIgnoreCase("ROLE_ONC")) {
+            username = roleOncUsername;
+            password = roleOncPassword;
+        } else if (role.equalsIgnoreCase("ROLE_ACB")) {
+            username = roleAcbUsername;
+            password = roleAcbPassword;
+        }
```
Comment (andlar) [inline src/test/java/gov/healthit/chpl/aqa/stepDefinitions/LoginSteps.java:77]:
```suggestion
        } else {
        throw new IllegalArgumentException("Incorrect role found: " + role);
        }
```
If you don't catch this here, we'd just end up waiting for the login page to load, then waiting for an empty username/password to be submitted, which probably wouldn't even work, but in any case we should do something about it

**Review by andlar — APPROVED** [review]

**Pull request merged**